crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -487,10 +488,16 @@ impl WritebackCache {
         //   the tx finalizer, plus checkpoint executor, consensus, and RPCs from fullnodes.
         let mut entry = self.dirty.objects.entry(*object_id).or_default();
 
-        self.cached.object_by_id_cache.insert(
-            object_id,
-            LatestObjectCacheEntry::Object(version, object.clone()),
-        );
+        self.cached
+            .object_by_id_cache
```

Comment:
Do you think it's worth adding a retry for the Write case?  Unlikely we hit the race twice in a row